src/database/sqlite/mod.rs

```diff
@@ -2130,7 +2216,167 @@ CREATE TABLE spend_transactions (
+        eprintln!("{}", tmp_dir.as_path().to_string_lossy());
+        fs::create_dir_all(&tmp_dir).unwrap();
+        let db_path: path::PathBuf = [tmp_dir.as_path(), path::Path::new("lianad_v3.sqlite3")]
+            .iter()
+            .collect();
+        let mut options = dummy_options();
+        options.schema = V3_SCHEMA;
+        options.version = 3;
+        create_fresh_db(&db_path, options, &secp).unwrap();
+
+        {
+            // Don't use SqliteDb::new() in order not to apply migration.
+            let db = SqliteDb {
+                db_path: db_path.clone(),
+            };
+            let mut conn = db.connection().unwrap();
+            assert!(conn.db_version() == 3);
+
+            let coin_a = Coin {
+                outpoint: bitcoin::OutPoint::from_str(
+                    "6f0dc85a369b44458eba3a1f0ea5b5935d563afb6994f70f5b0094e05be1676c:1",
+                )
+                .unwrap(),
+                is_immature: false,
+                block_info: None,
+                amount: bitcoin::Amount::from_sat(1231001),
+                derivation_index: bip32::ChildNumber::from_normal_idx(101).unwrap(),
+                is_change: false,
+                spend_txid: None,
+                spend_block: None,
+            };
+            let coin_b = Coin {
+                outpoint: bitcoin::OutPoint::from_str(
+                    "81b2f327d4c1fd67afd039374f8798fd9ff37932c6f5c221c1c569350eac5ac8:19234",
+                )
+                .unwrap(),
+                is_immature: false,
+                block_info: None,
+                amount: bitcoin::Amount::from_sat(23145),
+                derivation_index: bip32::ChildNumber::from_normal_idx(10).unwrap(),
+                is_change: false,
+                spend_txid: None,
+                spend_block: None,
+            };
+            let coin_c = Coin {
+                outpoint: bitcoin::OutPoint::from_str(
+                    "7477017f992cdc7ba08acafb77cb3b5bc0f42ac340d3e1e1da0785bdda20d5f6:932",
+                )
+                .unwrap(),
+                is_immature: false,
+                block_info: None,
+                amount: bitcoin::Amount::from_sat(354764),
+                derivation_index: bip32::ChildNumber::from_normal_idx(3401).unwrap(),
+                is_change: true,
+                spend_txid: None,
+                spend_block: None,
+            };
+            let coin_d = Coin {
+                outpoint: bitcoin::OutPoint::from_str(
+                    "ed6c8f1af9325f84de521e785e7ddfd33dc28c9ada4d687dcd3850100bde54e9:932",
+                )
+                .unwrap(),
+                is_immature: false,
+                block_info: None,
+                amount: bitcoin::Amount::from_sat(23200),
+                derivation_index: bip32::ChildNumber::from_normal_idx(3401).unwrap(),
+                is_change: true,
+                spend_txid: None,
+                spend_block: None,
+            };
```

Comment:
Same here. This one is the same as coin c. Maybe make it being unconfirmed but spent by an unconfirmed tx by setting `spend_txid`?